common/src/primitives/ip_address.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+/// IPv4 address
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Decode, Encode)]
```

Comment:
Am I right that we need these types to implement `Encode` and `Decode` for them? Would it make sense to add a documentation note explaining why we need almost exact copies of the std-types?..

Reply:
Yes, correct, to be able implement `Encode` and `Decode`. Documentation added, thanks.